x-pack/legacy/plugins/infra/public/containers/logs/log_analysis/log_analysis_setup_state.tsx

```diff
@@ -17,38 +20,19 @@ type SetupHandler = (
   endTime: number | undefined
 ) => void;
 
-export type ValidationIndicesUIError =
-  | ValidationIndicesError
-  | { error: 'NETWORK_ERROR' }
-  | { error: 'TOO_FEW_SELECTED_INDICES' };
-
-interface ValidIndex {
-  validity: 'valid';
-  name: string;
-  isSelected: boolean;
-}
-
-interface InvalidIndex {
-  validity: 'invalid';
-  name: string;
-  errors: ValidationIndicesError[];
-}
-
-export type ValidatedIndex = ValidIndex | InvalidIndex;
-
 interface AnalysisSetupStateArguments<JobType extends string> {
-  cleanupAndSetupModule: SetupHandler;
```

Comment:
_bikeshed_: I get what you're going for here but I wonder if it makes things more confusing.

For example, in this line we do `..SetUpModule` but not `SetUpHandler`. A future reader might wonder if this is intentional or if it was an oversight. If the type refers as "the hander of the setup", or "the handler that sets up", but has a typo on the name.

I did a quick grep and most of kibana uses `setup` as a verb as well as a noun. I know it's not gramatically correct, but maybe it's more pragmatic.

Reply:
I carefully changed it like that to explicitly avoid the confusion, because I figured a reader would assume intent when reading code :wink:

If it is `setUpHandler`, it is a function that sets up a handler, because "to set up" is the verb. When it is `SetupHandler` it is a handler function that handles the setup, because "setup" is a noun.

Is there a better way to disambiguate? Would calling it `ModuleSetupHandler` reduce the similarity and thereby the confusion?

Reply:
I think the confusion might come not from having `SetupHandler`, but from having `cleanUpAndSetUpModule`. I agree with you it's more correct, but it's also the only place where it's spelled correctly.

...but as I write that I realise it's not really an argument to leave it with the wrong spelling :)

In any case it's not a blocker